src/python-common/ceph/deployment/service_spec.py

```diff
@@ -1193,7 +1193,17 @@ def __init__(self,
         self.kmip_cert = kmip_cert
         self.kmip_key = kmip_key
         self.kmip_ca_cert = kmip_ca_cert
-        self.kmip_host_list = kmip_host_list
+        self.kmip_host_list: list[Dict[str, Union[str, int]]] = []
+        if isinstance(kmip_host_list, list):
+            # convert kmip host list of str to list of dict
+            if len(kmip_host_list) and isinstance(kmip_host_list[0], str):
+                self.kmip_host_list = [
+                    {'addr': host}
+                    for host in kmip_host_list
+                    if isinstance(host, str)
+                ]
+            else:
+                self.kmip_host_list = [host for host in kmip_host_list if isinstance(host, dict)]
```

Comment:
`kmip_host_list` can only be a list of strings, list of dicts, or `None`, correct? I feel like we could just do
```
self.kmip_host_list: List[Dict[str, str] = []
if kmip_host_list:
    for host_obj in kmip_host_list:
        if isinstance(host_obj, str):
            self.kmip_host_list.append({'addr': host_obj})
        elif isinstance(host_obj, dict):
            self.kmip_host_list.append(host_obj)
```
and maybe throw some validation error if we get something that isn't a list or a list that has something other that strings or dicts. What I posted I think would also allow a mixed format of strings and dicts